src/shared/ExtensionMessage.ts

```diff
@@ -102,73 +103,95 @@ export interface ExtensionMessage {
 	error?: string
 }
 
-export interface ApiConfigMeta {
-	id: string
-	name: string
-	apiProvider?: ApiProvider
-}
-
-export interface ExtensionState {
+export type ExtensionState = Pick<
```

Comment:
The `ExtensionState` can mostly be generate by `Pick`ing from `GlobalSettings`, but there are some annoying differences that we can eventually clean up and make this type look tidy.